File: project/.sdk/src/cmp/php/fragment/EntityCreateOp.fragment.php
require_once __DIR__ . '/../utility/struct/Struct.php';
require_once __DIR__ . '/../core/Helpers.php';

// EJECT-START

    public function create($reqdata, $ctrl = null): array
    {
        $utility = $this->_utility;
        $ctx = ($utility->make_context)([
            "opname" => "create",
            "ctrl" => $ctrl,
            "match" => $this->_match,
            "data" => $this->_data,
            "reqdata" => $reqdata,
        ], $this->_entctx);

        return $this->_run_op($ctx, function () use ($ctx) {
            if ($ctx->result) {
                if ($ctx->result->resdata) {
                    $this->_data = ProjectNameHelpers::to_map(Struct::clone($ctx->result->resdata)) ?? [];
                }
            }
        });
    }

// EJECT-END

File: project/.sdk/src/cmp/py/fragment/EntityCreateOp.fragment.py
from utility.voxgig_struct import voxgig_struct as vs
from core import helpers

entity_create_op = None

# EJECT-START

    def create(self, reqdata, ctrl=None):
        utility = self._utility
        ctx = utility.make_context({
            "opname": "create",
            "ctrl": ctrl,
            "match": self._match,
            "data": self._data,
            "reqdata": reqdata,
        }, self._entctx)

        def post_done():
            if ctx.result is not None:
                if ctx.result.resdata is not None:
                    self._data = helpers.to_map(vs.clone(ctx.result.resdata)) or {}

        return self._run_op(ctx, post_done)

# EJECT-END
